render-test/runner.cpp

```diff
@@ -322,14 +316,22 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
 
 bool TestRunner::checkProbingResults(TestMetadata& metadata) {
     if (metadata.metrics.isEmpty()) return true;
+    const auto writeMetrics = [&metadata](const mbgl::filesystem::path& path,
+                                          const std::string& message = std::string()) {
+        if (!metadata.metrics.isEmpty()) {
+            mbgl::filesystem::create_directories(path);
+            mbgl::util::write_file(path / "metrics.json", serializeMetrics(metadata.metrics));
+            metadata.errorMessage += message;
+        } else {
+            metadata.errorMessage += " Collected metrics are empty.";
```

Comment:
It won't end up with this case since the first line `if (metadata.metrics.isEmpty()) return true;` directly return, I assume this line should be moved to `bool checkResult = metadata.metrics.isEmpty() || checkFileSize(metadata) && checkMemory(metadata) && checkNetwork(metadata) &&
                       checkFps(metadata) && checkGfx(metadata);`